src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,638 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { useMergedRefs } from '../utils/hooks/useMergedRefs';
+import { useTheme } from '../utils/hooks/useTheme';
+import '@itwin/itwinui-css/css/slider.css';
+import { Tooltip } from '../Tooltip';
+import { Placement } from 'tippy.js';
+import { StylingProps } from '../utils/props';
+
+/** Specifies domain of allowable values. */
+export interface ValueDomain {
+  min: number;
+  max: number;
+}
+
+/** Determines which segments are shown with color.
+ * 'none' - no colored tracks are displayed.
+ * 'odd-segments'- colored tracks shown in segments 1,3,5, etc. Default if number of thumbs values are odd.
+ * 'even-segments'- colored tracks shown in segments 2,4,6, etc. Default if number of thumbs values are even.
+ * 'undefined' - temporary only here because storybook sends in string 'undefined' and not value undefined
+ */
+export type TrackDisplayMode =
+  | 'none'
+  | 'odd-segments'
+  | 'even-segments'
+  | 'undefined';
+
+/** The interaction mode. Default is 1. Possible values:
+ * 'allow-crossing' - allows thumb to cross other thumbs. Default.
+ * 'step-separated'- keeps the thumb from crossing and separated by a step.
+ */
+export type ThumbMoveMode = 'allow-crossing' | 'step-separated';
+
+/** Local Context set up by Slider and provided to children */
+interface SliderContextProps {
+  readonly values: number[];
+  readonly step: number;
+  readonly trackDisplayMode: TrackDisplayMode;
+  readonly railDomain: ValueDomain;
+  readonly thumbMode: ThumbMoveMode;
+  readonly toolTipPosition: Placement;
+  readonly setThumbMoveIndex: (index: number) => void;
+  readonly setThumbValue: (index: number, value: number) => void;
+  readonly toolTipFunction?: (val: number, step: number) => string;
+  readonly hideTooltip: boolean;
+}
+
+const SliderContext = React.createContext<SliderContextProps>({
+  values: [],
+  step: 1,
+  trackDisplayMode: 'none',
+  hideTooltip: false,
+  railDomain: { min: 0, max: 100 },
+  thumbMode: 'allow-crossing',
+  toolTipPosition: 'top',
+  setThumbMoveIndex: () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  },
+  setThumbValue: () => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  },
+});
+
+const useSliderContext = (): SliderContextProps => {
+  return React.useContext(SliderContext);
+};
+
+/** Private Utility Functions */
+function clamp(val: number, min: number, max: number) {
+  return val > max ? max : val < min ? min : val;
```

Comment:
Something like this could also work?
I used the same in `ProgressIndicators` and `Wizard`, just naming was different..
`Math.min(max, Math.max(min, val))`.
Maybe it would make sense to move it to our utils folder, maybe in common file and reuse it... `getBoundedValue(...)`